docs/release-strategy.md

```diff
@@ -68,3 +68,29 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
 1. Announce release via the following:
     - The `#announce` channel on Slack
     - The `announce` mailing list
+
+## Release notes
+
+To ensure our users are well-informed about new features, improvements, and breaking changes, we
+maintain a `PendingReleaseNotes.md` file. This file serves as a centralized place to document changes
+that will be included in the next major (X) release.
+
+### Editing Release Notes
+
+When submitting a Pull Request (PR) that introduces notable features or breaking changes, committers
+need to update the `PendingReleaseNotes.md` file. Clearly describe the changes, their impact, and
+any actions users might need to take. We want clear, concise, and user-friendly notes.
+
+### Branching for a New Release
+
+Each time we prepare for a new major (X) release, we branch out from the main codebase.
+As part of this branching process, the contents of `PendingReleaseNotes.md` are reviewed and
+finalized.
+
+### Resetting the Release Notes
```

Comment:
@leseb why not just get rid of the overhead of clearing `PendingReleaseNotes.md` and have the file being a running list of breaking changes with those for the upcoming release at the top. This was how it was done in Ceph (https://github.com/ceph/ceph/blob/main/PendingReleaseNotes). 

Though this file can get long I think there's value to having all of the breaking changes between versions in one place.

Reply:
+1

and in that case, I'd rename the file to `ReleaseNotes.md`, though `RELEASE_NOTES.md` would be more consistent with other files in the same directory.

Reply:
After some more digging, it looks like most of the notable open-source projects out there use a `CHANGELOG.md` file. The projects maintain a single `CHANGELOG.md` file that documents all releases. I like the suggestion @alimaredia @russellb, I think this approach provides a great balance between consistency, ease of management, and centralized information. Thanks!